### PR TITLE
[16.0][OU-ADD] hr_holidays

### PR DESCRIPTION
**Pull request by remytms**
Compute new values for allocation_validation_type on hr.leave.type as it is now a stored computed field with only two choices: "no" and "officer".

Load noupdate_changes.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
/ocabot migration hr_holidays

<dependency></dependency>

File: openupgrade_scripts/scripts/hr_holidays/16.0.1.5/post-migration.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openupgradelib import openupgrade
+
+
+def force_recompute_allocation_validation_type(env):
+    hr_leave_type_ids = (
+        env["hr.leave.type"].with_context({"active_test": False}).search([])
+    )
+    hr_leave_type_ids._compute_allocation_validation_type()
```
Comment (pedrobaeza) [inline openupgrade_scripts/scripts/hr_holidays/16.0.1.5/post-migration.py]: This is not performant and can lead to unexpected changes, triggering other computes including this one in depends, so it's advisable to do the equivalent one in SQL.

Comment (remytms) [reply openupgrade_scripts/scripts/hr_holidays/16.0.1.5/post-migration.py]: Thank you for the advice. It's now done by SQL directly. :)

File: openupgrade_scripts/scripts/hr_holidays/16.0.1.5/post-migration.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+from openupgradelib import openupgrade
+
+
+def set_allocation_validation_type(env):
+    """Operate like the _compute_allocation_validation_type() function.
+
+    It set "no" by default except if employee_request is set to "no"
+    where it set "officer".
+    """
+    openupgrade.logged_query(
```
Comment (pedrobaeza) [inline openupgrade_scripts/scripts/hr_holidays/16.0.1.5/post-migration.py:13]: You only need to reallocate those with `set` value to the corresponding one. Please don't write the rest that are OK.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: openupgrade_scripts/scripts/hr_holidays/16.0.1.5/post-migration.py (hunk 2 above)
Comment (pedrobaeza) [reply openupgrade_scripts/scripts/hr_holidays/16.0.1.5/post-migration.py:13]: As further explanation, imagine there's an extra module adding a new validation type. You are overwriting the leave types with this new validation type in this SQL.

**Comment by pedrobaeza** [comment]
Superseded by #4335 correcting the SQL query about `allocation_validation_type`, and also doing the annotation in the coverage file, so the migration scripts are also tested.

**Pull request closed without merging**